library/src/main/java/com/sothree/slidinguppanel/SlidingUpPanelLayout.java

```diff
@@ -1033,10 +1035,14 @@ private boolean isViewUnder(View view, int x, int y) {
     private int computePanelTopPosition(float slideOffset) {
         int slidingViewHeight = mSlideableView != null ? mSlideableView.getMeasuredHeight() : 0;
         int slidePixelOffset = (int) (slideOffset * mSlideRange);
+
```

Comment:
my guess is race condition / slower hardware, the logic IS exactly the same, but the jittery (and buggy) sliding happened because the return value would every now and then switch from X to -X
// adding this line already solved the problem
Log.d("jack", "values= " + getMeasuredHeight() + " " + getPaddingBottom() + " " + mPanelHeight + " " + slidePixelOffset );

when I added that line there for troubleshooting, the problem immediately disappeared, and seeing as the only thing happening was the 'pre-calculation' of several functioned values, having the values calculated before the actual return statement solved the strange behavior (I considered that a better 'solution' then leaving a debug line in)

Also as it turned out, HTC desire (2.2) had the problem, another alcatel one touch (with 2.3) didn't so it might be a bit more device specific, seeing as the logic doesn't change, this will probably make it work on more (possibly slower) devices

** topBoundCollapsed would be -660 instead of the expected 660 it did on 'slower slides' or just hitting DragView
private float computeSlideOffset(int topPosition) {
        // Compute the panel top position if the panel is collapsed (offset 0)
        final int topBoundCollapsed = computePanelTopPosition(0);

I agree it's been baffling me for a few days already :) but the above explanation seems to make the most sense, I do know that adding that line makes the library use-able (and sliding smooth) for me on my test device while without it the sliding is all over the place (-X here meant it wasn't sure if it wanted to be Top or Bottom Panel :) )